api/consent.go

```diff
@@ -0,0 +1,61 @@
+package api
+
+import (
+	"time"
+
+	"github.com/prebid/go-gdpr/consentconstants"
+)
+
+// VendorConsents is a GDPR Vendor Consent string, as defined by IAB Europe. For technical details,
+// see https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/Consent%20string%20and%20vendor%20list%20formats%20v1.1%20Final.md#vendor-consent-string-format-
+type VendorConsents interface {
+	// The version of the Consent string.
+	Version() uint8
+
+	// The time that the consent string was first created
+	Created() time.Time
+
+	// The time that the consent string was last updated
+	LastUpdated() time.Time
+
+	// The ID of the CMP used to update the consent string.
+	CmpID() uint16
+
+	// The version of the CMP used to update the consent string
+	CmpVersion() uint16
+
+	// The number of the CMP screen where consent was given
+	ConsentScreen() uint8
+
+	// The two-letter ISO639-1 language code used by the CMP to ask for consent, in uppercase.
+	ConsentLanguage() string
+
+	// The VendorListVersion which is needed to interpret this consent string.
+	//
+	// The IAB is hosting these on their webpage. For example, version 2 of the
+	// Vendor List can be found at https://vendorlist.consensu.org/v-2/vendorlist.json
+	//
+	// For other versions, just replace the "v-*" path with the value returned here.
+	// The latest version can always be found at https://vendorlist.consensu.org/vendorlist.json
+	VendorListVersion() uint16
+
```

Comment:
I see `api/consent.go` declares a number of fields that are consistent with the elements of the consent string elements listed in:
https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/Consent%20string%20and%20vendor%20list%20formats%20v1.1%20Final.md#vendor-consent-string-format-
            
Which, from the URL I infer is stil version `1.1` isn't it? I'm probably wonrg, but don't we need to update this fields to the TCF 2 consent string listed in this URL instead?
https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20Consent%20string%20and%20vendor%20list%20formats%20v2.md#the-core-string

Out of the consent strings listed in tables found in both URL's the fields are almost identical up to a certain point. In particular, TCF 2 seems to have different fields from `VendorListVersion() uint16` on.

Reply:
the tcf1 directory has the code for the tcf1 fields, the tcf2 directory code for tcf2 fields. tcf2 is basically tcf1 plus extras, so first pass is just to extract the fields that exist in both versions. Will likely define a second interface which impliments the first plus extra methods for the tcf2 only fields.